src/routes/checkout/Address/Address.js

```diff
@@ -1,5 +1,329 @@
-import React from "react";
+import React, { useState, useContext, useEffect, useRef } from "react";
 
+import { onAuthStateChanged } from "firebase/auth";
+import { AppContext } from "../../../context/app-context";
+import { UserContext } from "../../../context/user/user-context";
+import { SET_CHECKOUT_ADDRESS } from "../../../constant";
+
+import { doc, getDoc, setDoc } from "firebase/firestore";
+
+import {
+  ThemeProvider,
+  Container,
+  Typography,
+  Grid,
+  Autocomplete,
+  TextField,
+  Button,
+  CircularProgress,
+} from "@mui/material";
+import { ButtonStyles, responsiveTheme, textFieldStyles } from "./muiStyles";
 export default function Address() {
-  return <div>Address</div>;
+  const { dispatch: userDispatch } = useContext(UserContext);
+  const { auth, db } = useContext(AppContext);
+
+  const [loading, setLoading] = useState(false);
+
+  // Reference values of inputs
+  const buildingRef = useRef();
+  const apartmentNumberRef = useRef();
+  const messageRef = useRef();
+
+  useEffect(() => {
+    onAuthStateChanged(auth, async (user) => {
+      const userData = await getDoc(doc(db, "users", user.uid));
+
+      // Upon login, check for shipping_address data
+      if (userData.shipping_address) {
+        // Set shipping_address data from database in user context
+        userDispatch({
+          type: SET_CHECKOUT_ADDRESS,
+          payload: {
+            shipping_address: userData.shipping_address,
+          },
+        });
+      } else {
+        // Set new shipping_address entry for user in database
+        let shippingData = {
+          shipping_address: {
+            campus: "",
+            building: "",
+            floor_apartment: "",
+          },
+        };
+        await setDoc(userData, shippingData, { merge: true });
+      }
+    });
+  }, []);
+
+  // User submits new shipping address data
+  const handleSubmit = async () => {
+    const usersRef = doc(db, "users", auth.currentUser.uid);
```

Comment:
The logic to handle the address should be put inside a handler, in this case, `profile-context.js`. The reason why you would want to do this is to make the logic "separated". Inside the component itself, you only want to see the UI logic like display this, show that, and don't want to go into detail about how to update the database or how to fetch the necessary data. This can make future change easier, as people only need to make change to the handler instead of digging around the components